griffon/services/core_queries.py

```diff
@@ -129,30 +127,19 @@ def execute(self, status=None) -> dict:
         results = list()
         product_versions = set()
         product_streams = set()
-        with concurrent.futures.ThreadPoolExecutor() as executor:
-            futures = []
-            for affect in affects:
-                futures.append(
-                    executor.submit(
-                        self.corgi_session.components.retrieve_list,
-                        name=affect.ps_component,
-                        latest_components_by_streams=True,
-                        include_fields="product_streams.name,product_versions.name",
-                    )
-                )
-            for future in concurrent.futures.as_completed(futures):
-                try:
-                    for c in future.result().results:
-                        results.append(c.to_dict())
-                except Exception as exc:
-                    logger.error("%r generated an exception: %s" % (future, exc))
-                    exit(0)
-
-            for c in results:
-                for ps in c["product_streams"]:
-                    product_streams.add(ps["name"])
-                for pv in c["product_versions"]:
-                    product_versions.add(ps["name"])
+        for affect in affects:
+            components = self.corgi_session.components.retrieve_list_iterator_async(
+                name=affect.ps_component,
+                latest_components_by_streams="True",
```

Comment:
Nitpick: boolean values are handled internally by the bindings, so you may just supply python `True` value

Reply:
this might be a binding bug eg. when this is set to bool True we get

```
TypeError: Invalid variable type: value should be str, int or float, got True of type <class 'bool'>
```